ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -419,15 +421,13 @@ private void defineColumns(StringBuilder query) {
     }
     List<FieldSchema> cols = sourceTab.getSd().getCols();
     boolean isFirst = true;
+    List<String> columnDescs = new ArrayList<String>();
     for (FieldSchema col : cols) {
-      if (!isFirst) {
-        query.append(", ");
-      }
-      isFirst = false;
-      query.append("`").append(col.getName()).append("` ");
-      query.append(crud ? ":" : "");
-      query.append(col.getType());
+      String columnType = DDLPlanUtils.formatType(TypeInfoUtils.getTypeInfoFromTypeString(col.getType()));
+      String columnDesc = "`" + col.getName() + "`" + (crud ? ":" : "") + columnType;
```

Comment:
should we have an extra space after the column name in case of insert-only?

Reply:
We can add the extra space, honestly it won't make much of difference because this would be a temporary table and would be dropped after compaction.